silkworm/silkrpc/json/block.cpp

```diff
@@ -138,7 +138,7 @@ struct GlazeJsonBlock {
 
 struct GlazeJsonBlockReply {
     std::string_view jsonrpc = kJsonVersion;
-    uint32_t id;
+    std::variant<std::uint32_t, std::shared_ptr<std::string>> id;
```

Comment:
This type definition should be refactored as:
- `std::uint32_t` -> `uint32_t`
- `std::shared_ptr<std::string>` -> `std::string`

Moreover, I suggest to extract an explicit `using` directive not to repeat the same definition:

`using JsonRpcId = std::variant<uint32_t, std::string>;`